### PR TITLE
feat: Handle call timeout - Part 7 (AR-1480)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1480" title="AR-1480" target="_blank">SubtaskAR-1480</a>  handle call timeout
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

A follow up PR of #527 that aims to dismiss the calling screen for these cases:
- A calls B and B didn't answer the call with the 30s
- A calls B and A decided to end the call before B answer it
- A and B are in ongoing call, one of them ends the call

**Note that, if A calls B and B rejects the calls then the calling screen on A is not dismissed for PRIVACY reasons**

### Dependencies (Optional)

Needs this [feat: Initiate call - Part 6 (AR-1250)](https://github.com/wireapp/wire-android-reloaded/pull/527) 
and [feat: update call status when starting a call ](https://github.com/wireapp/kalium/pull/448)

Needs releases with:

- [ ] GitHub link to other pull request

#### How to Test

Select a conversation and try the usecases above

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3202108)(3202108)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3202108)(3202108)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3202210)(3202210)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3202210)(3202210)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3209595)(3209595)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3209595)(3209595)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3210143)(3210143)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3210143)(3210143)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt [hunk 1]
```diff
@@ -38,8 +43,27 @@ class InitiatingCallViewModel @Inject constructor(
...
+            if (it.first().conversationId == conversationId)
+                when (it.first().status) {
+                    CallStatus.CLOSED -> navigateBack()
+                    CallStatus.ESTABLISHED -> {
+                        navigateBack()
+                        navigationManager.navigate(
+                            command = NavigationCommand(
+                                destination = NavigationItem.OngoingCall.getRouteWithArgs(listOf(conversationId))
+                            )
+                        )
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt:63]: Maybe we could move this to a private function

**Review by alexandreferris — COMMENTED** [review]
Great work done here @ohassine !

Just a small comment :)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-533/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3216719)(3216719)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3216719)(3216719)-dev-release.apk)

**Pull request merged**